### PR TITLE
Allow user-friendly wake up / sleep time format.

### LINKED ISSUE #17: Can't set night owl schedule
I set the environment variables as follows
- `HUBOT_HEROKU_WAKEUP_TIME = '8:00'`
- `HUBOT_HEROKU_SLEEP_TIME = '2:00'`

but hubot puts `INFO Skipping keep alive, time to rest` log out and sleeps soon.
(I think [this line](https://github.com/hubot-scripts/hubot-heroku-keepalive/blob/master/src/heroku-keepalive.coffee#L57) decides the behavior.)

Does anyone know the way to avoid this problem?

### PR DESCRIPTION
**Pull request by treby**
# Overview
Currently, `HUBOT_HEROKU_WAKEUP_TIME` and `HUBOT_HEROKU_SLEEP_TIME` [expect  their relationship as `HUBOT_HEROKU_WAKEUP_TIME <  HUBOT_HEROKU_SLEEP_TIME` implicitly](https://github.com/hubot-scripts/hubot-heroku-keepalive/blob/8b38c6e35b6f5d63af76f06b77766c42c145485b/src/heroku-keepalive.coffee#L57). But sometimes we want to set wake up time and sleep time across midnight(e.g. `10:00` to `26:00` (`2:00` of next day)), and it seems no way to realize that.
This patch allows us to set bot active time around midnight by calculating a surplus of time offset.
Fix #17

# Implementation
Normalize sleep time and current time as distance of wake up time (`0 <= distance < 1,440(60 * 24)` [min]), then judge bot status by comparing these distances.
When "sleep time distance from wake up time" is larger than "current time distance from wake up time", it means _bot is still awake_, otherwise it tells to bot that _time to rest_.

# Testing
You can verify that the changes work right by simple test code below.

```coffeescript
isActive = (wakeUpTime, sleepTime, nowTime) ->
  wakeUpTime = wakeUpTime.split(':').map (i) -> i * 1
  sleepTime = sleepTime.split(':').map (i) -> i * 1
  nowTime = nowTime.split(':').map (i) -> i * 1
  
  wakeUpOffset = (60 * wakeUpTime[0] + wakeUpTime[1]) % (60 * 24)
  awakeMinutes = (60 * (sleepTime[0] + 24) + sleepTime[1] - wakeUpOffset) % (60 * 24)
  
  elapsedMinutes = (60 * (nowTime[0] + 24) + nowTime[1] - wakeUpOffset) % (60 * 24)
  elapsedMinutes < awakeMinutes

console.log isActive('6:00', '22:00', '0:00')   # false
console.log isActive('6:00', '22:00', '5:59')   # false
console.log isActive('6:00', '22:00', '6:00')   # true
console.log isActive('6:00', '22:00', '21:59')  # true
console.log isActive('6:00', '22:00', '22:00')  # false

console.log isActive('22:00', '6:00', '0:00')   # true
console.log isActive('22:00', '6:00', '5:59')   # true
console.log isActive('22:00', '6:00', '6:00')   # false
console.log isActive('22:00', '6:00', '21:59')  # false
console.log isActive('22:00', '6:00', '22:00')  # true

console.log isActive('8:00', '2:00', '0:00')    # true
console.log isActive('8:00', '2:00', '1:59')    # true
console.log isActive('8:00', '2:00', '2:00')    # false
console.log isActive('8:00', '2:00', '7:59')    # false
console.log isActive('8:00', '2:00', '8:00')    # true
```

### CODE REVIEW

**Comment by ESPNMichaelKidd** [comment]
:+1:

**Comment by davidwessman** [comment]
:+1:

**Comment by kamui** [comment]
:+1:

**Comment by seanjuarez** [comment]
:+1:

**Comment by SammyIsra** [comment]
To be clear, this is not yet on the main branch, correct? As of today, you need to have  HUBOT_HEROKU_WAKEUP_TIME < HUBOT_HEROKU_SLEEP_TIME, right?

**Comment by davidwessman** [comment]
Yep.
You can however use
```
HUBOT_HEROKU_WAKEUP_TIME = 08:00
HUBOT_HEROKU_SLEEP_TIME = 26:00
```
To sleep from 02:00-08:00.

**Comment by SammyIsra** [comment]
That's the format I recently changed it to, but the bot does not seem to be able to stay connected to the Slack channel. Maybe there's something that I'm doing wrong? Also, would this be the place to ask for troubleshooting help?

**Comment by davidwessman** [comment]
A better place would be stackoverflow.
Check your Heroku logs, and make sure that your scheduled startup is working.

**Comment by jgkim** [comment]
:+1: Any update on this, @technicalpickles?

**Comment by technicalpickles** [comment]
No update, sorry. Definitely need more help maintaining this script.

**Comment by treby** [comment]
If there are something I can do for this p-r, feel free to tell me :)

File: src/heroku-keepalive.coffee [hunk 1]
```diff
@@ -23,11 +23,11 @@
 #   Josh Nichols <technicalpickles@github.com>
 
 module.exports = (robot) ->
-  wakeUpTime = (process.env.HUBOT_HEROKU_WAKEUP_TIME or '6:00').split(':')
-  sleepTime = (process.env.HUBOT_HEROKU_SLEEP_TIME or '22:00').split(':')
+  wakeUpTime = (process.env.HUBOT_HEROKU_WAKEUP_TIME or '6:00').split(':').map (i) -> i * 1
```
Comment (technicalpickles) [inline src/heroku-keepalive.coffee]: Is this just converting them to integers? Might be able to use `parseInt` instead.

**Comment by technicalpickles** [comment]
I think some of time logic could use a refactoring to be a bit more clear (variable names, temporary variables) and tests to make sure it works as expected, but that is a problem already in the code, so wont' let that block getting this fix out.

I'm going to merge this & release it as a patch version since it can be considered a bug. @treby I'm adding you as a collaborator for this contribution, and if you send me your npm username, I can add you there too.

File: src/heroku-keepalive.coffee (hunk 1 above)
Comment (treby) [inline src/heroku-keepalive.coffee]: Yes, it seems more smart (and I didn't know the way). I'll fix the method to use `parseInt` thanks!

**Comment by treby** [comment]
@technicalpickles 
Thank you! I agree with what you say, and I'm going to improve the detail.
And my npm username is also [@treby](https://www.npmjs.com/~treby).

**Comment by treby** [comment]
@technicalpickles Thank you for adding npm package collaborator!
I've fixed the code that you mentioned and it seems to be for release of the patch version now. Can I do this?
If you have any problems such as variable names for the current code, tell me the point please.
(It's difficult to find out for me...) Thanks!

**Comment by davidwessman** [comment]
:+1:

**Comment by technicalpickles** [comment]
@treby I'd strongly recommend always using pull requests, just so there's some visibility for change (I had to look at the master branch to see there was a commit). We're using grunt for releasing, and since it's a bug fix, it'd be `grunt release:patch`.

**Comment by treby** [comment]
@technicalpickles I understood. I'll be careful to use pull requests from next time. And I see how to release. I'm going to release the patch version by using grunt.

**Comment by treby** [comment]
`grunt release:patch` done.

**Pull request merged**